presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkRemoteSourceOperator.java

```diff
@@ -74,18 +81,29 @@ public Page getOutput()
             return null;
         }
 
-        SerializedPage serializedPage = null;
+        PageBuilder pageBuilder = new PageBuilder(types);
         synchronized (iterator) {
-            if (iterator.hasNext()) {
-                serializedPage = iterator.next();
+            while (iterator.hasNext() && !pageBuilder.isFull()) {
+                PrestoSparkRow row = iterator.next();
```

Comment:
As a future work, I guess you want to use Spark's `MutableRow`? :) . cc @sameeragarwal

Reply:
Actually I'm not sure if we want to integrate with Spark's types at this point.